### ISSUE TITLE
Titlebar rendering bug when mousing over window buttons

### ISSUE THREAD

**Issue by zdavidsen** [microsoft/terminal#13904]
### Windows Terminal version

1.14.2281.0

### Windows build number

10.0.19044.0

### Steps to reproduce

Move the terminal window to secondary monitor (4K 150%)  (main monitor is 1080p 125%)
Mouse over window buttons (close, maximize, minimize)

### Expected Behavior

Each buttons highlights as the mouse enters, and unhighlights as the mouse exits

### Actual Behavior

The highlight appears to redraw every frame, reapplying the highlight on top of the existing highlight.  
Additionally, the highlight drawing follows the mouse across the title bar, even outside of the buttons.
Even stranger, the highlight seems to take some cue from pixel color behind the mouse.  I can sometimes turn the buttons blue by moving the mouse out of the terminal window and onto my (blue) desktop.

Unfortunately, I'm having trouble actually capturing evidence of this. I've tried 2 different screen recording softwares, and the highlighting looks completely normal when recorded.  I'll try posting a video from my phone once this is opened..

**Comment by zadjii-msft** [comment]
Yea I think we're gonna need the tape here to see what's going on 😉

**Comment by zdavidsen** [comment]
Apologies for the truly awful video quality

https://user-images.githubusercontent.com/22267435/188008957-bd53587b-93df-42ec-b53e-b524ca5f08df.mp4

**Comment by zadjii-msft** [comment]
That is... absolutely unlike anything I've ever seen before. Like, the DWM buttons are sometimes bleeding through? That's... totally not something we've seen before. What else do you have installed/? Maybe there's something like fancyzones or who knows what that's trying to hijack the caption buttons and fighting with us to draw them. Maybe it's a bad graphics driver issue? (though, I'd suspect the same thing would affect like, Teams or Spotify or anyone who draws custom titlebars)

**Comment by zdavidsen** [comment]
Yeah so that's part of the weirdness.. The only (known) thing I have installed that affects rendering is f.lux, and the problem still happens even when f.lux is disabled.  It also only happens on my external monitor, works entirely as expected on the built-in laptop display

**Comment by zdavidsen** [comment]
Some more info that may or may not be helpful:
- I don't have fancyzones or any other compositors/window managers/desktop modifiers installed (other than f.lux anyway)
- Other custom titlebars seem to be fine (teams, word, chrome, etc.)
- The two displays are running on different graphics hardware... Internal display is on Intel UHD 630, external display is on Nvidia Quadro P1000
- Today it's significantly less annoying... In office vs. at home.  Same external monitor model, but over display port over thunderbolt instead of hdmi.  I'll have to play around with that configuration a bit and see how much that affects it.

**Comment by zadjii-msft** [comment]
I honestly have no idea what we could possibly be doing that would cause that. Maybe it has to do with our use of a swap chain, with XAML on top of our caption buttons. Maybe it'll get better (or worse) once #13935 merges 🤷 

I also have no idea how we'd go about debugging this at this point - I'm highly worried that this is just something that's specific to your hardware setup 😕

**Comment by zdavidsen** [comment]
One more bit of info that I can't believe I left out: This is a relatively recent issue, and only started happening sometime in the last couple updates (not sure which exactly).

And yeah, I wouldn't be at all surprised if this is either extremely rare or specific to my setup.  I have an unfortunate tendency to attract this kind of extremely rare issue...

**Comment by yo1dog** [comment]
Not sure if this is related, but this happens whenever I resize the window:
![image](https://user-images.githubusercontent.com/1504597/190032753-e017069b-0cc8-4895-abb5-fb87dba7554e.png)

**Comment by zdavidsen** [comment]
@yo1dog is that happening for just a few frames while you resize, or is it a persistent effect?  I see that when I resize as well, but only in (roughly) the area that has been expanded since the last frame was rendered, and then it immediately fills in.

**Comment by yo1dog** [comment]
same: just a few frames.

**Comment by daroot** [comment]
I have a similar problem.  It's really hard to capture since attempting to screen shot causes a refresh of the screen which fixes it.

It's not just the window buttons for me, but the whole right quarter of the bar or so, though it's most often noticeable only on the buttons.  And not just hovering over it, but any time the focus changes, especially to other windows.  I'm using FancyZones and have the terminal on the left 2/3rds of the screen, and raising focus on the browser window on the right 1/3rd tends to mess up the last button on the Terminal window.  It's also not consistent as to exactly what distortion happens.  Sometimes its all buttons, sometimes it's just rightmost, sometimes it's full right half the close button, sometimes it's just the bottom of that button.

Of note, I also have two monitors with different resolutions.  And in my case, one is ultrawide landscape, and the other is portrait mode.  Both are 100% scale and are connected to the same video card (Nvidia GTX 1080).  It does not seem to matter which monitor the terminal window is on.  I can also confirm it does not happen on other windows with custom titlebars such as Chrome or Discord, only Terminal.

https://user-images.githubusercontent.com/1639570/195804324-316402c9-f737-49e8-9d92-602d77677fbd.mp4
